common/connector.go

```diff
@@ -124,3 +128,100 @@ type LightningConnector interface {
 	// connector.
 	FundsAvailable() (decimal.Decimal, error)
 }
+
+// PaymentStore is used to store payments for external services (except
+// exchange, which is explicitly notified) for give them ability to
+// check whether payment is received.
+type PaymentsStore interface {
+	// Payment returns payment by id
+	Payment(id string) (*Payment, error)
+	// AddPayment add payment to the store
+	AddPayment(p *Payment) error
+}
+
+var PaymentNotFound = errors.New("payment not found")
+
+// MemoryPaymentsStore is a `PaymentStore` in-memory implementation,
+// which is able to periodically clean stored payments which are exceed
+// storing time.
+type MemoryPaymentsStore struct {
+	paymentsMutex    sync.RWMutex
+	payments         map[string]paymentWithDatetime
+	storingTime      time.Duration
+	cleanerStopper   chan struct{}
+	cleanerWaitGroup sync.WaitGroup
+}
+
+func NewMemoryPaymentsStore(storingTime time.Duration) *MemoryPaymentsStore {
+	return &MemoryPaymentsStore{
+		payments:    map[string]paymentWithDatetime{},
+		storingTime: storingTime,
+	}
+}
+
+type paymentWithDatetime struct {
+	*Payment
+	datetime time.Time
+}
+
+func (s *MemoryPaymentsStore) Payment(id string) (*Payment, error) {
+	s.paymentsMutex.RLock()
+	defer s.paymentsMutex.RUnlock()
+	p, exists := s.payments[id]
+	if !exists {
+		return nil, PaymentNotFound
+	}
+	return p.Payment, nil
+}
+
+func (s *MemoryPaymentsStore) AddPayment(p *Payment) error {
```

Comment:
Comment on all exporter out of package methods

Reply:
fixed